engines/stark/ui/menu/locationscreen.cpp

```diff
@@ -171,6 +178,8 @@ void StaticLocationWidget::onClick() {
 
 	if (_soundMouseClick) {
 		_soundMouseClick->play();
+		// Ensure the click sound is played completely
+		while (_soundMouseClick->isPlaying()) {};
```

Comment:
It's not good to busy-wait for something to complete. The game appears unresponsive while waiting and the CPU is used at 100%.
In game engine design it's best to have a single game loop that roughly checks for user input, updates the simulated world and then renders. So far, Stark has managed to follow that rule. So here, we'll need to find another solution to let the sound fully play. Perhaps something is stopping the sound, preventing it from completing while it should not. Perhaps this code needs to wait for the sound to complete by checking each frame if it is finished.